src/main.rs

```diff
@@ -41,27 +50,62 @@ async fn main() -> anyhow::Result<()> {
 
     let mut pod_list = args.pods;
 
-    if !args.deployments.is_empty() {
-        for deploy in args.deployments.iter() {
-            pod_list.append(
-                &mut k8s::get_pod_list_for_deployment(&client, deploy, &args.namespace).await?,
-            );
-        }
+    enum ResourceType<'a> {
+        Deployment(&'a str),
+        StatefulSet(&'a str),
+        DaemonSet(&'a str),
+        Job(&'a str),
     }
 
-    if !args.statefulsets.is_empty() {
-        for statefulset in args.statefulsets.iter() {
-            pod_list.append(
-                &mut k8s::get_pod_list_for_statefulset(&client, statefulset, &args.namespace)
+    let mut resources = Vec::new();
+
+    resources.extend(
+        args.deployments
+            .iter()
+            .map(|deploy| ResourceType::Deployment(deploy)),
+    );
+    resources.extend(
+        args.statefulsets
+            .iter()
+            .map(|statefulset| ResourceType::StatefulSet(statefulset)),
+    );
+    resources.extend(args.daemonsets.iter().map(|ds| ResourceType::DaemonSet(ds)));
+    resources.extend(args.jobs.iter().map(|job| ResourceType::Job((job))));
```

Comment:
🚫 **[clippy]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> 
<pre><code>error: consider removing unnecessary double parentheses
  --> src/main.rs:73:67
   |
73 |     resources.extend(args.jobs.iter().map(|job| ResourceType::Job((job))));
   |                                                                   ^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#double_parens
   = note: `-D clippy::double-parens` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::double_parens)]`
</code></pre>
<!-- __reviewdog__:ChBkNDQ3NGRiNDJkMzNiZjZhEgZjbGlwcHk= -->